src/averageAges.js

```diff
@@ -55,7 +71,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let childrenWithMothers = people.filter(person => person.mother !== null);
+
+  if (onlyWithSon) {
+    childrenWithMothers = childrenWithMothers
+      .filter(child => child.sex === 'm');
+  }
+
+  let ageSum = 0;
+  let ageDifferenceCount = 0;
+
+  childrenWithMothers.forEach(child => {
+    people.forEach(person => {
+      if (person.name === child.mother) {
+        ageSum += child.born - person.born;
+        ageDifferenceCount++;
+      }
+    });
+  });
```

Comment:
can you do it without loops nesting? maybe split it into 2 blocks? Also, try using `reduce` here too